test/compilable/issue18097.d

```diff
@@ -7,6 +7,6 @@ unittest // this first unittest is needed to trigger the bug
 
 unittest // second unittest
 {
-    auto a = &mixin(__traits(identifier, __traits(parent, {  })));
+    auto a = __traits(identifier, __traits(parent, {  }));
```

Comment:
This isn't even remotely the same. Before it was a function pointer, now it's a string.

Reply:
Should have added a comment that this was just to see if the other test passed. Indeed that change shouldn't go in as-is.